### PR TITLE
JAMES-2813 Bind SerializedTaskManager to CassandraEventStore

### PR DESCRIPTION
**Pull request by blackheaven**

### CODE REVIEW

File: event-sourcing/event-store-cassandra/src/test/java/org/apache/james/eventsourcing/eventstore/cassandra/CassandraEventStoreExtension.java [hunk 1]
```diff
@@ -37,13 +38,19 @@
...
-    private final Set<EventDTOModule> modules;
-    private CassandraClusterExtension cassandra;
+    private final CassandraClusterExtension cassandra;
+    private final Supplier<JsonEventSerializer> serializerSupplier;
     private EventStoreDao eventStoreDao;
 
     public CassandraEventStoreExtension(@SuppressWarnings("rawtypes") EventDTOModule... modules) {
-        this.modules = Arrays.stream(modules).collect(ImmutableSet.toImmutableSet());
+        @SuppressWarnings("rawtypes")
+        Set<EventDTOModule> savedModules = Arrays.stream(modules).collect(ImmutableSet.toImmutableSet());
```
Comment (mbaechler) [inline event-sourcing/event-store-cassandra/src/test/java/org/apache/james/eventsourcing/eventstore/cassandra/CassandraEventStoreExtension.java]: ImmutableSet.copyOf should work as well

Comment (mbaechler) [reply event-sourcing/event-store-cassandra/src/test/java/org/apache/james/eventsourcing/eventstore/cassandra/CassandraEventStoreExtension.java]: what about  `EventDTOModule<?, ?>` ?

File: event-sourcing/event-store-cassandra/src/test/java/org/apache/james/eventsourcing/eventstore/cassandra/CassandraEventStoreExtension.java [hunk 2]
```diff
@@ -37,13 +38,19 @@
...
+        serializerSupplier = () -> new JsonGenericEventSerializer(savedModules);
+        this.cassandra = new CassandraClusterExtension(CassandraEventStoreModule.MODULE);
+    }
+
+    public CassandraEventStoreExtension(Supplier<JsonEventSerializer> serializerSupplier) {
```
Comment (mbaechler) [inline event-sourcing/event-store-cassandra/src/test/java/org/apache/james/eventsourcing/eventstore/cassandra/CassandraEventStoreExtension.java]: you should not have two constructors doing field initialization. Only one and others must call it.

File: server/container/guice/guice-common/pom.xml [hunk 3]
```diff
@@ -37,6 +37,23 @@
             <artifactId>apache-mailet-test</artifactId>
             <scope>test</scope>
         </dependency>
+        <dependency>
+            <groupId>${james.groupId}</groupId>
+            <artifactId>apache-james-backends-cassandra</artifactId>
```
Comment (mbaechler) [inline server/container/guice/guice-common/pom.xml]: cassandra in guice-common ?

File: server/task-cassandra/src/main/java/org/apache/james/task/eventsourcing/cassandra/TasksSerializationModule.java [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+    Function<JsonTaskSerializer, EventDTOModule<Cancelled, DTO.CancelledDTO>> CANCELLED = jsonTaskSerializer -> EventDTOModule
+        .forEvent(Cancelled.class)
+        .convertToDTO(DTO.CancelledDTO.class)
+        .toDomainObjectConverter(dto -> dto.toDomainObject(jsonTaskSerializer))
+        .toDTOConverter(TaskSerializer::fromDomainObject)
+        .typeName("task-manager-cancelled")
+        .withFactory(EventDTOModule::new);
+
+    Function<JsonTaskSerializer, List<EventDTOModule<? extends TaskEvent, ? extends DTO.TaskEventDTO>>> MODULES = jsonTaskSerializer -> ImmutableList
+        .of(CREATED, STARTED, CANCEL_REQUESTED, CANCELLED, COMPLETED, FAILED)
```
Comment (mbaechler) [inline server/task-cassandra/src/main/java/org/apache/james/task/eventsourcing/cassandra/TasksSerializationModule.java]: Stream.of should be simpler

File: server/task-cassandra/src/main/scala/org/apache/james/task/eventsourcing/cassandra/TasksSerialization.scala [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+
+import scala.collection.JavaConverters._
+import org.apache.james.eventsourcing.eventstore.cassandra.JsonEventSerializer
+import org.apache.james.eventsourcing.eventstore.cassandra.dto.{EventDTO, EventDTOModule}
+import org.apache.james.eventsourcing.{Event, EventId}
+import org.apache.james.server.task.json.JsonTaskSerializer
+import org.apache.james.server.task.json.dto.TaskDTO
+import org.apache.james.task.eventsourcing._
+import org.apache.james.task.{Task, TaskId}
+import play.api.libs.json._
```
Comment (mbaechler) [inline server/task-cassandra/src/main/scala/org/apache/james/task/eventsourcing/cassandra/TasksSerialization.scala]: could you explain how play json is articulated with JsonSerializer ?

Comment (mbaechler) [reply server/task-cassandra/src/main/scala/org/apache/james/task/eventsourcing/cassandra/TasksSerialization.scala]: Oh, the commits are messed-up in github at least, I think I know understand the design

File: server/task-cassandra/src/main/scala/org/apache/james/task/eventsourcing/cassandra/TasksSerialization.scala [hunk 6]
```diff
@@ -0,0 +1,150 @@
...
+
+object DTO {
```
Comment (mbaechler) [inline server/task-cassandra/src/main/scala/org/apache/james/task/eventsourcing/cassandra/TasksSerialization.scala]: I don't see any test?

File: event-sourcing/event-store-cassandra/src/main/java/org/apache/james/eventsourcing/eventstore/cassandra/JsonGenericEventSerializer.java [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+
+import org.apache.james.eventsourcing.Event;
+import org.apache.james.eventsourcing.eventstore.cassandra.dto.EventDTO;
+import org.apache.james.eventsourcing.eventstore.cassandra.dto.EventDTOModule;
+import org.apache.james.json.JsonGenericSerializer;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.google.common.collect.ImmutableSet;
+
+public class JsonGenericEventSerializer implements JsonEventSerializer {
```
Comment (mbaechler) [inline event-sourcing/event-store-cassandra/src/main/java/org/apache/james/eventsourcing/eventstore/cassandra/JsonGenericEventSerializer.java]: could I suggest JsonJacksonEventSerializer?

File: event-sourcing/event-store-cassandra/src/test/java/org/apache/james/eventsourcing/eventstore/cassandra/CassandraEventStoreExtension.java (hunk 1 above)
Comment (blackheaven) [reply event-sourcing/event-store-cassandra/src/test/java/org/apache/james/eventsourcing/eventstore/cassandra/CassandraEventStoreExtension.java]: it'll break a lot of code

File: event-sourcing/event-store-cassandra/src/test/java/org/apache/james/eventsourcing/eventstore/cassandra/CassandraEventStoreExtension.java (hunk 2 above)
Comment (blackheaven) [reply event-sourcing/event-store-cassandra/src/test/java/org/apache/james/eventsourcing/eventstore/cassandra/CassandraEventStoreExtension.java]: discarded

File: server/task-cassandra/src/main/java/org/apache/james/task/eventsourcing/cassandra/TasksSerializationModule.java (hunk 4 above)
Comment (blackheaven) [reply server/task-cassandra/src/main/java/org/apache/james/task/eventsourcing/cassandra/TasksSerializationModule.java]: discarded

File: event-sourcing/event-store-cassandra/src/main/java/org/apache/james/eventsourcing/eventstore/cassandra/JsonGenericEventSerializer.java (hunk 7 above)
Comment (blackheaven) [reply event-sourcing/event-store-cassandra/src/main/java/org/apache/james/eventsourcing/eventstore/cassandra/JsonGenericEventSerializer.java]: discarded

**Comment by blackheaven** [comment]
See #2559

**Pull request closed without merging**